### PR TITLE
[BPF] Do not apply normal *-hep policy to wep - align behaviour with iptables

### PR DESCRIPTION
**Pull request by tomastigera**
## Description

If there is no wildcard HEP, there is no policy that should be applied. But without skipping, empty list of profiles would create a default deny rule if none of the non-existent profiles matches. That is obviously always hit and traffic toward the host is dropped if defaultEndpointToHostAction is set to RETURN.

In fact, to be compatible with iptables behaviour, we should never apply normal *-hep policy, only the preDNAT one. This PR leaves the normal policy code there for enabling it in the future.

Also includes FV tests for defaultEndpointToHostAction by @fasaxc that codify behaviour and unify it across dataplanes.

Removed blanket ACCEPT in INPUT chain, only wg traffic is explicitly accepted now, the rest is dropped if INPUT policy is DROP and defaultEndpointToHostAction==RETURN.
## Related issues/PRs

fixes https://github.com/projectcalico/calico/issues/7252
## Todos

- [ ] Tests
- [ ] Documentation
- [ ] Release note

## Release Note

```release-note
ebpf: align defaultEndpointToHostAction with iptables - do not apply *-hep policy to wep
```

## Reminder for the reviewer

Make sure that this PR has the correct labels and milestone set.

Every PR needs one `docs-*` label.

- `docs-pr-required`: This change requires a change to the documentation that has not been completed yet.
- `docs-completed`: This change has all necessary documentation completed.
- `docs-not-required`: This change has no user-facing impact and requires no docs.

Every PR needs one `release-note-*` label.

- `release-note-required`: This PR has user-facing changes. Most PRs should have this label.
- `release-note-not-required`: This PR has no user-facing changes.

Other optional labels:

- `cherry-pick-candidate`: This PR should be cherry-picked to an earlier release. For bug fixes only.
- `needs-operator-pr`: This PR is related to install and requires a corresponding change to the operator.

### CODE REVIEW

**Comment by marvin-tigera** [comment]
OK, I will merge the pull request when it's ready, leave the commits as is when I merge it, and leave the branch after I've merged it.

**Comment by marvin-tigera** [comment]
Removing "merge-when-ready" label due to new commits

**Comment by marvin-tigera** [comment]
OK, I will merge the pull request when it's ready, leave the commits as is when I merge it, and leave the branch after I've merged it.

File: felix/fv/bpf_test.go
Comment (fasaxc) [inline felix/fv/bpf_test.go:634]: This is only right if the iptables INPUT chain policy is ACCEPT.  To be sure it's doing the right thing, we need to test both cases of the iptables policy.

Comment (tomastigera) [reply felix/fv/bpf_test.go:634]: yes, that is another step in fixing a broader issue

File: felix/dataplane/linux/bpf_ep_mgr.go
Comment (fasaxc) [inline felix/dataplane/linux/bpf_ep_mgr.go:1971]: Given what I discovered about iptables mode, shouldn't we _always_ suppress the normal HEP ingress policy?  We should render the HEP pre-DNAT policy only?

File: felix/dataplane/linux/int_dataplane.go
Comment (fasaxc) [inline felix/dataplane/linux/int_dataplane.go:1567]: What about traffic arriving on eth0, i.e. traffic to the host but not from a workload?  Will that get caught by this rule change?

Comment (tomastigera) [reply felix/dataplane/linux/int_dataplane.go:1567]: In that case, if INPUT is DROP it will get dropped unless it is wireguard. afaict it is what would happen in iptables. It is covered by the wg tests. If default hepToHost is ACCEPT, it will get accepted by other rules sooner.

File: felix/dataplane/linux/bpf_ep_mgr.go
Comment (tomastigera) [reply felix/dataplane/linux/bpf_ep_mgr.go:1971]: Is that case covered by the new fv test?

Comment (tomastigera) [reply felix/dataplane/linux/bpf_ep_mgr.go:1977]: wasn't covered, added those test cases

**Review by fasaxc — APPROVED** [review]

**Pull request merged**
